### DISCUSSION TITLE
My environment variables aren't working??

### DISCUSSION THREAD

**Discussion by DarshGupta16** [nuxt/nuxt#27467] [Questions] [answered]
Just as I said in the title. I've spent hours trying to find out what's wrong but I'm not able to figure it out. This is my `.env` file in the root of my project -

```
OWM_API_KEY=...
FIRE_API_KEY=...
```

This is nuxt.server.ts -
```
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  devtools: { enabled: true },
  postcss: {
    plugins: {
      tailwindcss: {},
      autoprefixer: {},
    },
  },
  css: ["~/assets/css/main.css"],
  runtimeConfig: {
    private: {
      fireApi: process.env.FIRE_API_KEY,
      owmApi: process.env.OWM_API_KEY,
    },
  },
});
```

And when I try this in the pages/index.vue file -
```
<script lang="ts" setup>
console.log(useRuntimeConfig().private.fireApi);
</script>
```

I get this error -
`[nuxt] error caught during app initialization TypeError: Cannot read properties of undefined (reading 'fireApi')`

Also one more thing to note is that I'm using VS Code and the environment variables I created in the `.env` file just don't show up in the intellisense. Any variables created in the `.env` file are supposed to be auto imported and made accessible in the `process.env` file as well right? Can someone help me please?

Another little interesting thing to note- the API key gets printed in the terminal. Not sure if that's gonna be very helpful, but I'll tell you anyway.

**Comment by group6digital** [comment, marked as answer]
You're exporting the private key to the client. Try adding a check like ```if(import.meta.server)``` to ensure that the private key is only accessible on the server-side.

From docs:
> Security note: Be careful not to expose runtime config keys to the client-side by either rendering them or passing them to useState.

**Reply by DarshGupta16** [reply]
Yeah! Thanks so much- I completely forgot about that. I'm able to access it in the api directory.
